src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,661 @@
+        if latents is not None:
+            if not torch.is_tensor(latents):
+                raise ValueError("`latents` must be a torch.FloatTensor.")
+            if not latents.dim() != 4:
+                raise ValueError(f"`latents` has unsupported dimensions or shape: {latents.shape}.")
+
+            if processing_resolution > 0:
+                max_orig = max(H, W)
+                new_H = H * processing_resolution // max_orig
+                new_W = W * processing_resolution // max_orig
+                if new_H == 0 or new_W == 0:
+                    raise ValueError(f"Extreme aspect ratio of the input image: [{W} x {H}]")
+                W, H = new_W, new_H
+            w = (W + self.vae_scale_factor - 1) // self.vae_scale_factor
+            h = (H + self.vae_scale_factor - 1) // self.vae_scale_factor
+            shape_expected = (num_images * ensemble_size, self.latent_space_size, h, w)
+
+            if latents.shape != shape_expected:
+                raise ValueError(f"`latents` has unexpected shape={latents.shape} expected={shape_expected}.")
+
+        # generator checks
+        if generator is not None:
+            device = self._execution_device
+            if isinstance(generator, torch.Generator):
+                if generator.device != device:
+                    raise ValueError("`generator` device differs from the pipeline's device.")
+            elif isinstance(generator, list):
+                if len(generator) != num_images * ensemble_size:
+                    raise ValueError(
+                        "The number generators must match the total number of ensemble members for all input images."
+                    )
+                if not all(g.device == device for g in generator):
+                    raise ValueError("At least one of the `generator` devices differs from the pipeline's device.")
+            else:
+                raise ValueError(f"Unsupported generator type: {type(generator)}.")
+
+    @torch.no_grad()
+    @replace_example_docstring(EXAMPLE_DOC_STRING)
+    def __call__(
+        self,
+        image: Union[Image.Image, np.ndarray, torch.FloatTensor],
+        num_inference_steps: Optional[int] = None,
+        ensemble_size: int = 1,
+        processing_resolution: Optional[int] = None,
+        match_input_resolution: bool = True,
+        resample_method_input: str = "bilinear",
+        resample_method_output: str = "bilinear",
+        batch_size: int = 1,
+        check_input: bool = True,
+        ensembling_kwargs: Optional[Dict[str, Any]] = None,
+        latents: Optional[Union[torch.FloatTensor, List[torch.FloatTensor]]] = None,
+        generator: Optional[Union[torch.Generator, List[torch.Generator]]] = None,
+        output_prediction_format: str = "np",
+        output_visualization: bool = True,
+        output_visualization_kwargs: Optional[Dict[str, Any]] = None,
+        output_uncertainty: bool = True,
+        output_latent: bool = False,
```

Comment:
```suggestion
```
let's combine this argument with `output_type` like we do with all our pipelines https://github.com/huggingface/diffusers/blob/69fdb8720ffdf3b0c629d5d2372032115b23c805/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L1044

it can be `np`, `pt`, `pil` and `latent`

Reply:
We currently already have a `.latent` field in the output. It is an important use-case to output both the `.prediction` values and the `.latent`. The latent output can be used to initialize the the `latents` pipeline kwarg when invoking the pipeline on the subsequent video frame. At the same time, the desired pipeline output is still the prediction itself.. An example of this use case can be found in the video processing tab here: https://huggingface.co/spaces/prs-eth/marigold-lcm

Reply:
ok:) thanks for explaining!